llvm/lib/Target/SPIRV/SPIRVGlobalRegistry.h

```diff
@@ -336,6 +347,7 @@ class SPIRVGlobalRegistry {
   MachineFunction *setCurrentFunc(MachineFunction &MF) {
     MachineFunction *Ret = CurMF;
     CurMF = &MF;
+    LastInsertedType = nullptr;
```

Comment:
The problem here would be when a user of this function switches contexts between machine functions. After this change the context would be not only `MachineFunction *` that is returned, but also last inserted type position. See, for example:

https://github.com/llvm/llvm-project/blob/372611e186fc7eaea4f94c3ee7516d5d0e655dcd/llvm/lib/Target/SPIRV/SPIRVISelLowering.cpp#L291-L295

Here the context is switched between function definition/function call, and we would reset `LastInsertedType` to nullptr in both to/fro cases.